### ISSUE TITLE
[feature] JSON output for `conan profile list`

### ISSUE THREAD

**Issue by monsdar** [conan-io/conan#5937]
Right now the command `conan profile list` does not have a `--json` option. We parse the output by manually. This is error prone and could easily change in the future. This issue is about adding a `--json` option like all the other commands have. It should simply output the installed profiles into a JSON file.

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by czoido** [comment]
Hi @monsdar,
Thanks for the feature request, we'll have a look at it

**Comment by mrunderline** [comment]
May I work on this issue?

**Comment by mrunderline** [comment]
I think it's almost done; but I prefer to check json schema with you, before send any PR!
The result of `conan profile show mynewprofile` command is as below:
```
Configuration for profile mynewprofile:

[settings]
os=Linux
os_build=Linux
arch=x86_64
arch_build=x86_64
compiler=gcc
compiler.version=8
compiler.libcxx=libstdc++11
[options]
MyLib:shared=True
[build_requires]
*: Tool1/0.1@user/channel, Tool2/0.1@user/channel, Tool3/0.1@user/channel, Tool4/0.1@user/channel
[env]
zlib:CC=/usr/bin/clang
zlib:CXX=/usr/bin/clang++
```
And created file with `conan profile list --json pro.json` command is like this:
```
{
  "mynewprofile": {
    "settings": {
      "os": "Linux",
      "os_build": "Linux",
      "arch": "x86_64",
      "arch_build": "x86_64",
      "compiler": "gcc",
      "compiler.version": "8",
      "compiler.libcxx": "libstdc++11"
    },
    "options": {
      "MyLib:shared": "True"
    },
    "build_requires": [
      {
        "*": [
          "Tool1/0.1@user/channel",
          "Tool2/0.1@user/channel",
          "Tool3/0.1@user/channel",
          "Tool4/0.1@user/channel"
        ]
      }
    ],
    "env": {
      "zlib:CC": "/usr/bin/clang",
      "zlib:CXX": "/usr/bin/clang++"
    }
  }
}
```

Is it acceptable?

**Comment by memsharded** [comment]
Hi @mrunderline 

Thanks very much for willing to contribute! I have assigned the issue to you then.

I think that @monsdar request is to get the actual list of profiles in a json format, not the contents. Like the list of names of the profiles file names. Is that correct @monsdar ?

**Comment by monsdar** [comment]
Correct, just a simple JSON array with the profile names would be enough
for me. It does not need to be more than what the commands prints to stdout.

James <notifications@github.com> schrieb am Sa., 19. Okt. 2019, 18:43:

**Comment by mrunderline** [comment]
Hi, It's done and PR sent!
Also I'll create a new PR to let people save profile detail as a json file.

**Comment by memsharded** [comment]
Merged! Will be released in Conan 1.20
